### PR TITLE
To Do list: add & remove: Milestone 3.

### PR DESCRIPTION
**Pull request by aliabbani**
## To-Do list: add & remove: Milestone 3.

In this milestone we:

- Removed all hardcoded items from the tasks array.
- Created a new JavaScript file.
- Implemented a function for adding a new task (add a new element to the array).
- Implemented a function for editing task descriptions.
- Implemented a function for deleting a task (remove an element from the array).
- Implemented a function for the "Clear all completed" button (use filter() method).
- Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique.
- All changes to the To-Do List should be saved in local storage.

### CODE REVIEW

File: src/addremove.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+const updatedIndex = () => {
+  tasks = JSON.parse(localStorage.getItem('tasks'));
+  for (let i = 0; i < tasks.length; i += 1) {
+    tasks[i].index = i + 1;
+  }
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+};
+
+export default updatedIndex;
```
Comment (descholar-ceo) [inline src/addremove.js:11]: :+1: This method works well kudos :muscle:

File: src/index.js [hunk 2]
```diff
@@ -103,3 +102,98 @@ const generateTasks = () => {
...
+  document.querySelector('.box').appendChild(div1);
+  listText.value = task.description;
+
+  const deleteIcon = document.createElement('i');
+  deleteIcon.className = 'fas fa-trash';
+  listText.addEventListener('keyup', () => {
+    task.description = listText.value;
+    threeDotIcon.replaceWith(deleteIcon);
+    localStorage.setItem('tasks', JSON.stringify(tasks));
+  });
```
Comment (descholar-ceo) [inline src/index.js]: - [ ] I could not find the button or link for deleting one task, would you add it?

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
Hi @aliabbani , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/addremove.js (hunk 1 above)
Comment (aliabbani) [reply src/addremove.js:11]: Thank you :)

File: src/index.js (hunk 2 above)
Comment (aliabbani) [reply src/index.js]: Hello Mugirase Emmanuel, Thank you for mentioning this. I added the button inside an addEventListener and you can see in the image below
tse remark

**Review by descholar-ceo — APPROVED** [review]
Hi @aliabbani , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**